test/jest/dep-graph-builders/yarn-lock-v2.test.ts

```diff
@@ -1,58 +1,444 @@
+    expect(dg.getPkgs()).toMatchObject([
+      { name: 'simple-cyclic-chain', version: '1.0.0' },
+      { name: 'a', version: '1.0.0' },
+      { name: 'b', version: '1.0.0' },
+    ]);
+  });
+
+  it('returns pkg deps - cyclic', async () => {
+    const opts: YarnLockV2ProjectParseOptions = {
+      includeDevDeps: false,
+      includeOptionalDeps: false,
+      strictOutOfSync: false,
+      pruneWithinTopLevelDeps: false,
+    };
+    const pkgJson = getHandRolledPkgJson('simple-cyclic-chain');
+    const yarnLock = getHandRolledYarnLock('simple-cyclic-chain');
+    const dg = await parseYarnLockV2Project(pkgJson, yarnLock, opts);
+    expect(dg.toJSON().graph.nodes).toMatchObject([
+      {
+        nodeId: 'root-node',
+        pkgId: 'simple-cyclic-chain@1.0.0',
+        deps: [{ nodeId: 'a@1.0.0' }],
+      },
+
+      {
+        nodeId: 'a@1.0.0',
+        pkgId: 'a@1.0.0',
+        deps: [{ nodeId: 'b@1.0.0' }],
+        info: { labels: { scope: 'prod' } },
+      },
+      {
+        nodeId: 'b@1.0.0',
+        pkgId: 'b@1.0.0',
+        deps: [{ nodeId: 'a@1.0.0' }],
+        info: { labels: { scope: 'prod' } },
+      },
+    ]);
+  });
+
+  it('returns packages - cyclic - pruned', async () => {
+    const opts: YarnLockV2ProjectParseOptions = {
+      includeDevDeps: false,
+      includeOptionalDeps: false,
+      strictOutOfSync: false,
+      pruneWithinTopLevelDeps: true,
+    };
+    const pkgJson = getHandRolledPkgJson('simple-cyclic-chain');
+    const yarnLock = getHandRolledYarnLock('simple-cyclic-chain');
+    const dg = await parseYarnLockV2Project(pkgJson, yarnLock, opts);
+    expect(dg.getPkgs()).toMatchObject([
+      { name: 'simple-cyclic-chain', version: '1.0.0' },
+      { name: 'a', version: '1.0.0' },
+      { name: 'b', version: '1.0.0' },
+    ]);
+  });
+
+  it('returns pkg deps - cyclic - pruned', async () => {
+    const opts: YarnLockV2ProjectParseOptions = {
+      includeDevDeps: false,
+      includeOptionalDeps: false,
+      strictOutOfSync: false,
+      pruneWithinTopLevelDeps: true,
+    };
+    const pkgJson = getHandRolledPkgJson('simple-cyclic-chain');
+    const yarnLock = getHandRolledYarnLock('simple-cyclic-chain');
+    const dg = await parseYarnLockV2Project(pkgJson, yarnLock, opts);
+    expect(dg.toJSON().graph.nodes).toMatchObject([
+      {
+        nodeId: 'root-node',
+        pkgId: 'simple-cyclic-chain@1.0.0',
+        deps: [{ nodeId: 'a@1.0.0' }],
+      },
+      {
+        nodeId: 'a@1.0.0',
+        pkgId: 'a@1.0.0',
+        deps: [{ nodeId: 'b@1.0.0' }],
+        info: { labels: { scope: 'prod' } },
+      },
+      {
+        nodeId: 'b@1.0.0',
+        pkgId: 'b@1.0.0',
+        deps: [{ nodeId: 'a@1.0.0:pruned' }],
+        info: { labels: { scope: 'prod' } },
+      },
+      {
+        nodeId: 'a@1.0.0:pruned',
+        pkgId: 'a@1.0.0',
+        deps: [],
+        info: { labels: { scope: 'prod', pruned: 'true' } },
```

Comment:
is it pruned: true or cyclic https://github.com/snyk/dep-graph/blob/master/src/core/types.ts#L51 do we not really care?

Reply:
I no longer care